rdrf/rdrf/views/patients_listing.py

```diff
@@ -91,10 +90,26 @@ def build_context(self):
             "facets": self.facets
         }
 
-    def get_configure_columns(self, columns):
-        for i, col in enumerate(columns):
-            col.configure(self.registry_model, self.user, i)
-        return [col for col in columns if col.user_can_see]
+    def get_user_table_config(self, column_to_dict=False):
+        registry_config = PatientListConfiguration(self.registry_model)
+        registry_columns = registry_config.get_columns()
+        registry_facets = registry_config.get_facets()
+
+        # initialise data table columns
+        for i, (column_key, datatable_column) in enumerate(registry_columns.items()):
+            datatable_column.configure(self.registry_model, self.user, i)
+
+        user_columns_dict = {key: val for key, val in registry_columns.items() if val.user_can_see}
+        user_facets = {key: val for key, val in registry_facets.items() if key in user_columns_dict.keys()}
+        user_columns = user_columns_dict.values()
+
+        if column_to_dict:
```

Comment:
Agree it makes more sense in `build_context`. 

I found the general layout and logic to the code in this class quite difficult to follow and it would probably benefit from a refactor. I tried to fix as much as I could that related to this change. Something for another time...

Reply:
Updated in the latest commit.